lib/cn/mbset.c

```diff
@@ -189,44 +161,43 @@ mbset_create(
 
     /* one allocation for:
      * - the mbset struct
-     * - array of mblock ids
-     * - array of udata structs
+     * - array of kvs_mblk_desc
+     * - array of udata elements
      */
-    alloc_len = sizeof(*self) + (sizeof(*self->mbs_idv) * idc) + (udata_sz * idc);
+    alloc_len  = sizeof(*self);
+    alloc_len += idc * sizeof(*self->mbs_mblkv);
+    alloc_len += idc * udata_sz;
 
     self = calloc(1, alloc_len);
     if (!self)
         return merr(ev(ENOMEM));
 
-    self->mbs_idv = (void *)(self + 1);
-    self->mbs_udata = (void *)(self->mbs_idv + idc);
-
-    assert(((void *)self) + alloc_len == (void *)(self->mbs_udata + idc * udata_sz));
-
-    memcpy(self->mbs_idv, idv, sizeof(*idv) * idc);
+    self->mbs_mblkv = (void *)(self + 1);
+    self->mbs_udata = (void *)(self->mbs_mblkv + idc);
 
-    self->mbs_idc = idc;
+    self->mbs_mblkc = idc;
     self->mbs_mp = mp;
     self->mbs_del = false;
     self->mbs_udata_sz = udata_sz;
 
-    err = mbset_map(self, flags);
-    if (ev(err))
+    err = mbset_mmap(self, idc, idv);
+    if (err)
         goto fail;
 
-    /* Must have mapped mblocks prior to this b/c the udata
-     * callback uses the maps */
-    err = mbset_mblk_getprops(self, udata_init_fn);
-    if (ev(err))
-        goto fail;
+    for (uint i = 0; i < idc; i++) {
+        err = udata_init(&self->mbs_mblkv[i], mbset_get_udata(self, i));
+        if (ev(err))
+            goto fail;
+    }
```

Comment:
I hate mbsets.  Want to get rid of them.  The had more value when mcache maps were kernel based.  They had less value when mcache maps moved to user space, and even less value without mache maps.

Reply:
Can you create a ticket for this? Maybe a 3.1 thing!